app/jobs/concerns/maintenance_tasks/task_job_concern.rb

```diff
@@ -135,8 +135,20 @@ def on_shutdown
         @task.run_callbacks(:interrupt)
       end
 
+      db_lock_version = @run.class.where(id: @run.id).pluck(:lock_version).first
+      if db_lock_version != @run.lock_version
+        raise ActiveRecord::StaleObjectError, @run
+      end
+
       @run.cursor = cursor_position
       @ticker.persist
+
+      # Refresh lock version here since we've written to the run by
+      # persisting progress
+      refresh_run_lock_version
```

Comment:
Can we update the lock_version in memory in `Run#persist_progress` instead, allowing the next update to succeed?